internal/controller/prefectserver_controller.go

```diff
@@ -203,9 +204,23 @@ func (r *PrefectServerReconciler) reconcileMigrationJob(ctx context.Context, ser
 		condition = conditions.AlreadyExists(objName, errorMessage)
 
 		return &ctrl.Result{}, errors.NewBadRequest(errorMessage)
-	} else if jobNeedsUpdate(&foundMigrationJob.Spec, &desiredMigrationJob.Spec, log) {
-		// TODO: handle replacing the job if something has changed
+	} else if migrationJobNeedsUpdate(&foundMigrationJob.Spec, &desiredMigrationJob.Spec, log) {
+		log.Info("migration Job needs to be updated and replaced", "name", desiredMigrationJob.Name)
+		condition = conditions.NeedsUpdate(objName)
 
+		// deletes the Job, so it can be recreated on the following reconciliation.
+		// k8s Jobs are immutable and need to be recreated if they are to be re-run.
+		if foundMigrationJob.GetDeletionTimestamp() == nil {
+			log.Info("Deleting existing migration Job", "name", foundMigrationJob.Name)
+			if err = r.Delete(ctx, foundMigrationJob); err != nil {
+				condition = conditions.NotDeleted(objName, err)
+				return &ctrl.Result{}, err
+			}
+			condition = conditions.Deleted(objName)
+		} else {
+			log.Info("Waiting for migration Job to be deleted, so it can be recreated", "name", desiredMigrationJob.Name)
+		}
+		return &ctrl.Result{Requeue: true}, nil
```

Comment:
there's no "waiting" in the operator pattern, so we have to make this reconciliation handler idempotent. so:
- if we detect that an update (actually, a replacement) is needed
- we'll take the existing Job resource, see if it's been marked for deletion -> if not, delete it
- we can't be sure of how many reconciliations it will take, so we'll assume it takes more than 1
- we'll continue to requeue the reconciliation job until the `foundMigrationJob` is cleaned up, so that the very next run will create a new Job

Reply:
https://sdk.operatorframework.io/docs/faqs/#how-do-i-wait-for-some-specific-cluster-state-such-as-a-resource-being-deleted